src/LabLog/Startup.cs

```diff
@@ -43,7 +43,7 @@ public void Configure(IApplicationBuilder app, IHostingEnvironment env)
             {
                 routes.MapRoute(
                     name: "default",
-                    template: "{controller=Admin}/{action=Index}/{id?}");
+                    template: "{controller=Admin}/{action=Index}/{id?}/{name?}");

```

Comment:
Rather than using imperative mapping like this, you can do it in a more declarative way.
On top of your controllers and actions you can have Route attributes. I'll add to yours as example